include/boost/geometry/strategies/geographic/distance_cross_track.hpp

```diff
@@ -103,128 +144,106 @@ class geographic_cross_track
-            result.closest_point_lat = lat;
-        }
-        return result;
-    }
-
-    template <typename CT>
-    result_distance_point_segment<CT>
-    static inline non_iterative_case(CT const& lon1, CT const& lat1, //p1
-                                     CT const& lon2, CT const& lat2, //p2
-                                     Spheroid const& spheroid)
-    {
-        CT distance = geometry::strategy::distance::geographic<FormulaPolicy, Spheroid, CT>
-                              ::apply(lon1, lat1, lon2, lat2, spheroid);
-
-        return non_iterative_case(lon1, lat1, distance);
-    }
-
-    template <typename CT>
-    CT static inline normalize(CT const& g4, CT& der)
+    auto static inline normalize(CT const& g4, CT& der)
     {
         CT const pi = math::pi<CT>();
         if (g4 < -1.25*pi)//close to -270
         {
 #ifdef BOOST_GEOMETRY_DEBUG_GEOGRAPHIC_CROSS_TRACK
-            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to -270" << std::endl;
+            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to -270"
+                      << std::endl;
 #endif
             return g4 + 1.5 * pi;
         }
         else if (g4 > 1.25*pi)//close to 270
         {
 #ifdef BOOST_GEOMETRY_DEBUG_GEOGRAPHIC_CROSS_TRACK
-            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to 270" << std::endl;
+            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to 270"
+                      << std::endl;
 #endif
             der = -der;
             return - g4 + 1.5 * pi;
         }
         else if (g4 < 0 && g4 > -0.75*pi)//close to -90
         {
 #ifdef BOOST_GEOMETRY_DEBUG_GEOGRAPHIC_CROSS_TRACK
-            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to -90" << std::endl;
+            std::cout << "g4=" << g4 * math::r2d<CT>() <<  ", close to -90"
+                      << std::endl;
 #endif
             der = -der;
             return -g4 - pi/2;
         }
         return g4 - pi/2;
     }
 
+
     template <typename CT>
     static void bisection(CT const& lon1, CT const& lat1, //p1
                           CT const& lon2, CT const& lat2, //p2
                           CT const& lon3, CT const& lat3, //query point p3
                           Spheroid const& spheroid,
                           CT const& s14_start, CT const& a12,
-                          result_distance_point_segment<CT>& result)
+                          result_type<CT>& result)
     {
-        typedef typename FormulaPolicy::template direct<CT, true, false, false, false>
-                direct_distance_type;
-        typedef typename FormulaPolicy::template inverse<CT, true, false, false, false, false>
-                inverse_distance_type;
+        using direct_distance_type = typename FormulaPolicy::template direct
+            <
+                CT,
+                true,
+                false,
+                false,
+                false
+            >;
```

Comment:
I propose to put all of this in one line. I can't see any benefits of formatting it like this.